src/storage.h

```diff
@@ -206,28 +228,179 @@ class Storage
                       "uilabel":"Batch Handling"}
   bool discrete_handling;
 
-  #pragma cyclus var {"default": 1,\
-                      "tooltip": "Length of the active buying "\
-                        "period",\
-                      "doc":"During the length of the active buying "\
-                        "period, agent exhibits regular behavior. "\
-                        "If paired with dormant buying period, "\
-                        "alternates between buying and not buying, "\
-                        "regardless if space is available",\
-                      "uilabel":"Active Buying Period"}
-  int active_buying;
+  #pragma cyclus var {"default": "Fixed",\
+                      "tooltip": "Type of active buying frequency",\
+                      "doc": "Options: Fixed, Uniform, Normal. Fixed requires active_buying_val. Uniform "\
+                      "requires active_buying_min and active_buying_max.  Normal "\
+                      "requires active_buying_mean and active_buying_std, with optional "\
+                      "active_buying_min and active_buying_max.",\
+                      "uitype": "combobox",\
+                      "categorical": ["Fixed", "Uniform", "Normal"],\
+                      "uilabel": "Active Buying Frequency Type"}
+  std::string active_buying_frequency_type;
+
+  #pragma cyclus var {"default": -1,\
+                      "tooltip": "Fixed active buying frequency",\
```

Comment:
What would be the difference between active and dormant buying?

Reply:
Active is default behavior, agent will register demand if space is available. Dormant is no buying at all, regardless of space in inventory